spec.md

```diff
@@ -2770,8 +3198,9 @@ type_ground = "Clock" | "Reset" | "AsyncReset"
             | ( "UInt" | "SInt" | "Analog" ) , [ width ] ;
 type_aggregate = "{" , field , { field } , "}"
                | type , "[" , int , "]" ;
+type_ref = ( "Probe" | "RWProbe" ) , "<", type , ">" ;
 field = [ "flip" ] , id , ":" , type ;
-type = [ "const" ], ( type_ground | type_aggregate ) ;
+type = ( [ "const" ] , ( type_ground | type_aggregate ) ) | type_ref ;
```

Comment:
Is it mentioned anywhere above that probes cannot be constant?

Reply:
Yes, should clarify that a ref is already const so no reason to mark it as such.  And that cannot get a `RWProbe` to a const thing.

I'm making a note to do another pass with const in mind, now that this has come up.  It was merged right as I was submitting the PR so the integration of the spec language may have been a little hasty, thanks/apologies.  However I think this is right (but should be made clear so you don't have to gather it by inspecting the grammar! :laughing:):

References are always "const", arguably "more const than const" in that they /must/ be known statically (not only that they cannot mutate at "runtime").  I suppose we could allow someone to redundantly mark them "const" (similarly, we may choose to accept "const const  UInt" as harmless), don't feel strongly about that at all.

References /to/ const are supported, and as mentioned above I should clarify you cannot get a RWProbe to anything const! Thanks :smile:.

Reply:
@dtzSiFive and I discussed this some.  On one hand, forcing a constant seems to be non-constanty.  On the other hand, verification is probably going to want to do that anyway.